pkg/upgrade/upgrader/cstor_cspc.go

```diff
@@ -60,7 +61,11 @@ func NewCSPCPatch(opts ...CSPCPatchOptions) *CSPCPatch {
 
 // PreUpgrade ...
 func (obj *CSPCPatch) PreUpgrade() error {
-	err := obj.CSPC.PreChecks(obj.From, obj.To)
+	err := isOperatorUpgraded("cspc-operator", obj.Namespace, obj.To, obj.KubeClientset)
```

Comment:
Should it go into precheck right(because it is mostly like check before upgrade)?

Reply:
Precheck is for a specific CR, it does not checks for any other resource. Preupgrade is for a volume or pool as a whole(means openebs CRs + deploy + service), so it makes more sense to have it in preupgrade.